lib/src/main/ktx/com/fabernovel/statefullayout/StatefulLayout.kt

```diff
@@ -106,3 +109,62 @@ fun StatefulLayout.showLoading(): State = showState(R.id.stateLoading)
  * @return the content state
  */
 fun StatefulLayout.showContent(): State = showState(R.id.stateContent)
+
+/**
+ * Sets a minimal time to wait after each successful state change triggered by [requestStateChange].
+ *
+ * All requests triggered from [requestStateChange] will be handled in respect of this minimal
+ * time, sometimes dropping some requests when many (backpressure).
+ *
+ * Default value for [timeToWaitInMillis] is [DEFAULT_REGULATOR_PACE]
+ *
+ * @see [StateChangeRegulator] for details about regulation rules.
+ */
+@FlowPreview
+@ExperimentalCoroutinesApi
+fun StatefulLayout.setMinimalTimeToWaitBetweenStateChanges(timeToWaitInMillis: Long) {
```

Comment:
Valid for `StatefulLayout.requestStateChange(...)` and `StatefulLayout.stateChangeRegulator(...)` too.

Reply:
Might be better indeed to rely on a user provided coroutine scope instead of created one specific to our view which could lead to un-wanted behaviours 👍

Reply:
I see two other options:
- We expose 2 suspend functions: no need for local coroutine scope but we gotta be very clear about the fact that `setMinimalTimeToWaitBetweenStateChanges` is going to suspend forever and never return (calls flow.collect {}) = should be called from a dedicated coroutine... Might not be the best developer experience.
- We use fragment's scope: Getting the host fragment with [`View.findFragment()`](https://developer.android.com/reference/kotlin/androidx/fragment/app/package-summary#(android.view.View).findFragment()) then using `fragment.viewLifecycleOwner.lifecycleScope`. Problem is, the view is not guaranteed to be attached to a fragment = we risk an `IllegalStateException`.

Reply:
In the `StateRegulator` handleStateChangeRequests could launch a separate coroutine to collect requests ?

This way `setMinimalTimeToWaitBetweenStarteChanges` won't suspend forever right ? 🤔

Reply:
Might work, but I guess we'd rather directly request a coroutineScope as a method argument instead of secretly seeking the `coroutineContext` and starting new  async coroutines (then why calling it suspend if it starts async work?)
> If you need to launch a coroutine that keeps running after your function returns, then make your function an extension of CoroutineScope or pass scope: CoroutineScope as parameter to make your intent clear in your function signature. Do not make these functions suspending
> https://link.medium.com/4Nzoib7zk8